selfdrive/test/process_replay/test_processes.py

```diff
@@ -1,117 +1,164 @@
+  if not os.path.isfile(cmp_log_fn):
+    req = requests.get(BASE_URL + os.path.basename(cmp_log_fn))
+    assert req.status_code == 200, ("Failed to download %s" % cmp_log_fn)
+
+    with tempfile.NamedTemporaryFile(suffix=".bz2") as f:
+      f.write(req.content)
+      f.flush()
+      f.seek(0)
+      cmp_log_msgs = list(LogReader(f.name))
+  else:
+    cmp_log_msgs = list(LogReader(cmp_log_fn))
+
+  log_msgs = replay_process(cfg, lr)
+
+  # check to make sure openpilot is engaged in the route
+  if cfg.proc_name == "controlsd" and FULL_TEST:
+    for msg in log_msgs:
+      if msg.which() == "controlsState":
+        if msg.controlsState.active:
+          break
+    else:
+      segment = cmp_log_fn.split("/")[-1].split("_")[0]
+      raise Exception("Route never enabled: %s" % segment)
+
+  ignore.extend(cfg.ignore)
+  return compare_logs(cmp_log_msgs, log_msgs, ignore)
+
+def format_diff(results, ref_commit):
+  diff1, diff2 = "", ""
+  diff2 += "***** tested against commit %s *****\n" % ref_commit
+
+  failed = False
+  for segment, result in list(results.items()):
+    diff1 += "***** results for segment %s *****\n" % segment
+    diff2 += "***** differences for segment %s *****\n" % segment
+
+    for proc, diff in list(result.items()):
+      diff1 += "\t%s\n" % proc
+      diff2 += "*** process: %s ***\n" % proc
+
+      if isinstance(diff, str):
+        diff1 += "\t\t%s\n" % diff
+        failed = True
+      elif len(diff):
+        cnt = {}
+        for d in diff:
+          diff2 += "\t%s\n" % str(d)
+
+          k = str(d[1])
+          cnt[k] = 1 if k not in cnt else cnt[k] + 1
+
+        for k, v in sorted(cnt.items()):
+          diff1 += "\t\t%s: %s\n" % (k, v)
+        failed = True
+  return diff1, diff2, failed
+
 if __name__ == "__main__":
 
-  process_replay_dir = os.path.dirname(os.path.abspath(__file__))
-  ref_commit_fn = os.path.join(process_replay_dir, "ref_commit")
+  parser = argparse.ArgumentParser(description="Regression test to identify changes in a process's output")
 
-  if not os.path.isfile(ref_commit_fn):
+  # whitelist has precedence over blacklist in case both are defined
+  parser.add_argument("--whitelist-procs", type=str, nargs="*", default=[],
+                        help="Whitelist given processes from the test (e.g. controlsd)")
+  parser.add_argument("--whitelist-cars", type=str, nargs="*", default=[],
+                        help="Whitelist given cars from the test (e.g. HONDA)")
+  parser.add_argument("--blacklist-procs", type=str, nargs="*", default=[],
+                        help="Blacklist given processes from the test (e.g. controlsd)")
+  parser.add_argument("--blacklist-cars", type=str, nargs="*", default=[],
+                        help="Blacklist given cars from the test (e.g. HONDA)")
+  parser.add_argument("--ignore-fields", type=str, nargs="*", default=[],
+                        help="Extra fields or msgs to ignore (e.g. carState.events)")
```

Comment:
Can I also ignore a certain log type entirely? eg: CarEvents.

Reply:
I thought the ignore fields would work for this but I guess it doesn't for carEvents when the list isn't initialized or the length is 0. I added an `--ignore-msgs` arg for this